aten/src/ATen/native/TensorProperties.cpp

```diff
@@ -60,30 +60,10 @@ Tensor contiguous(const Tensor& self, MemoryFormat memory_format) {
   if (self.is_contiguous(memory_format)) {
     return self;
   }
-  auto  result = at::empty_like(self);
-  switch (memory_format) {
-    case MemoryFormat::Any: // Back compatibility with old defaults
-    case MemoryFormat::Contiguous: {
-      break;
-    }
-    case MemoryFormat::ChannelsLast: {
-      AT_CHECK(
-          result.dim() == 4,
-          " required rank 4 tensor to use channels_last format");
-      std::vector<int64_t> newStrides(self.dim());
-      auto sizes = result.sizes();
-      newStrides[1] = 1;
-      newStrides[3] = sizes[1];
-      newStrides[2] = newStrides[3] * sizes[3];
-      newStrides[0] = newStrides[2] * sizes[2];
-      result = result.as_strided(sizes, newStrides);
-      break;
-    }
-    default: {
-      AT_CHECK(false, " unsupported memory format");
-    }
-  }
+  TORCH_CHECK(
+      memory_format != MemoryFormat::Preserve, " unsupported memory format");
```

Comment:
I don't think this error message is what you think it is

Reply:
In more words: when this fails, I will get the error message " unsupported memory format", full stop. You don't get the condition test here printed out. You'll need to write a more detailed error message here.

Reply:
Updated